### PR TITLE
Ports the mentor system from Hippie

### PR DESCRIPTION
**Pull request by Crossedfall**
[Changelogs]: # Direct port of Hippie's mentor system

:cl: optional name here
add: Mentor verbs and rank
config: Adds mentor config options
/:cl:

[why]: # Because there are mentors without any mentor tools.

### CODE REVIEW

File: DS13/code/modules/mentor/follow.dm [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	if(!is_mentor())
+		return
+	if(isnull(M))
+		return
+	if(!ismob(usr))
+		return
+	mentor_datum.following = M
+	usr.reset_perspective(M)
+	verbs += /client/proc/mentor_unfollow
+	to_chat(GLOB.admins, "<span class='mentor'><span class='prefix'>MENTOR:</span> <EM>[key_name(usr)]</EM> is now following <EM>[key_name(M)]</span>")
```
Comment (Kmc2000) [inline DS13/code/modules/mentor/follow.dm:11]: Dont think youve added this mentor span to the stylesheets for normal + dark mode chat themes

File: DS13/code/modules/mentor/follow.dm [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	to_chat(usr, "<span class='info'>Click the \"Stop Following\" button in the Mentor tab to stop following [key_name(M)].</span>")
+	log_mentor("[key_name(usr)] began following [key_name(M)]")
+
+/client/proc/mentor_unfollow()
```
Comment (Kmc2000) [inline DS13/code/modules/mentor/follow.dm:15]: Does this show up on rightclick? OR is it like FLW on tickets?

File: code/modules/admin/secrets.dm [hunk 3]
```diff
@@ -108,6 +109,11 @@
 				dat += "No-one has done anything this round!"
 			usr << browse(dat, "window=admin_log")
 
+		// DS13 - Mentor system -- access mentor log
+		if("mentor_log") 
+			MentorLogSecret() 
+		// DS13 - Mentor system end
```
Comment (Kmc2000) [inline code/modules/admin/secrets.dm]: DeepSpace13 - Mentor system (for consistency)

File: code/modules/client/client_procs.dm [hunk 4]
```diff
@@ -89,6 +89,11 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 		cmd_admin_pm(href_list["priv_msg"],null)
 		return
 
+	// hippie start -- Mentor PM
```
Comment (Kmc2000) [inline code/modules/client/client_procs.dm]: DeepSpace13 start

File: config/config.txt [hunk 5]
```diff
@@ -46,6 +46,14 @@ ADMIN_LEGACY_SYSTEM
 ##If enabled, each time admins are loaded ranks the database will be updated with the current ranks and their flags
 #LOAD_LEGACY_RANKS_ONLY
 
+## Comment this out if you want to use the SQL based mentor system, the legacy system uses mentors.txt.
```
Comment (Kmc2000) [inline config/config.txt:49]: Note to self: Add this on server box

**Review by Kmc2000 — CHANGES_REQUESTED** [review]

File: config/config.txt (hunk 5 above)
Comment (Crossedfall) [reply config/config.txt:49]: You'll also want to run the included SQL file to add the database tables.

File: code/modules/client/client_procs.dm (hunk 4 above)
Comment (Crossedfall) [reply code/modules/client/client_procs.dm]: Corrected.

File: code/modules/admin/secrets.dm (hunk 3 above)
Comment (Crossedfall) [reply code/modules/admin/secrets.dm]: Corrected

File: DS13/code/modules/mentor/follow.dm (hunk 1 above)
Comment (Crossedfall) [reply DS13/code/modules/mentor/follow.dm:11]: Added

File: DS13/code/modules/mentor/follow.dm (hunk 2 above)
Comment (Crossedfall) [reply DS13/code/modules/mentor/follow.dm:15]: Like FLW on tickets

**Comment by francinum** [comment]
@Kmc2000 DO NOT TESTMERGE PRs TAGGED `SQL` **WITHOUT EXPLICITLY NOTIFYING ME**

**Comment by Harry-Jubilee** [comment]
@crossedfall @francinum @Kmc2000 

So what exactly is happening with this?

**Comment by Harry-Jubilee** [comment]
Closed-

New system is being worked on, but thank you.

**Pull request closed without merging**